src/kit.ts

```diff
@@ -716,9 +737,13 @@ async function tryCreateNewVCEnvironment(inst: VSInstallation, hostArch: string,
   if (!variables) {
     return null;
   }
+  const major_version = majorVersionSemver(inst.installationVersion) ?? inst.installationVersion;
+  const arch = kitHostTargetArch(hostArch, targetArch);
+  const short_name = `msvc${major_version}_${arch}`;
```

Comment:
I tried to make default shortNames sensible, but I am not 100% sure about them. For example clang/gcc only use major version a no arch, but I have no expierence with cross-compilation, so I am not sure if that's good enough.

Reply:
We recently took a PR (#1200) to introduce compact and icon-only visibility modes for the status bar.  Maybe we don't need to do anything for this anymore?

Reply:
My use case for this is that I need build directories separated by kits, but the default buildKit command produces really long kit names (and makes them un-openable through console in vscode). I know one can create custom kits and give them shorter custom name, but I thought that having sensible build folders out of the box would be nice. #1200 seems to only resolve status bar length problems. Does changing kits-schema introduce some kind of breaking change? I saw some recent PR that tries to solve the kits-needing-refresh problem.

Reply:
> My use case for this is that I need build directories separated by kits, but the default buildKit command produces really long kit names

That makes sense.  We'll take a look at this and if we have more questions, we'll ask.  Just to set expectations though, we are testing the release build for 1.4 and plan to publish it in a day or so, so this won't make it into 1.4.  And since we won't release 1.5 for a few months after that and we have some other work we'll be starting that is not related to this extension, it may be a few weeks before we get to this PR.  Thank you in advance for understanding.